hamilton/function_modifiers/macros.py

```diff
@@ -1055,3 +1174,345 @@ def chain_transforms(
+        if self.collapse:
+            raise NotImplementedError(
+                "Collapsing functions as one node is not yet implemented for mutate(). Please reach out if you want this feature."
+            )
+        if self.chain:
+            raise NotImplementedError("@flow() is not yet supported -- this is ")
+
+        self.remote_applicables = tuple(
+            [apply_to(fn) if isinstance(fn, Callable) else fn for fn in target_functions]
+        )
+        self.mutating_function_kwargs = mutating_function_kwargs
+        self.restrict_to_single_module = True
+
+    def validate_same_module(self, mutating_fn: Callable):
+        """Validates target functions are in the same module as the mutator function.
+
+        :param mutating_fn: Function to validate against
+        :return: Nothing, raises exception if not valid.
+        """
+        local_module = mutating_fn.__module__
+        for remote_applicable in self.remote_applicables:
+            if remote_applicable.target_fn.__module__ != local_module:
+                raise NotSameModuleError(fn=mutating_fn, target_fn=remote_applicable.target_fn)
+
+    def _create_step(
+        self, mutating_fn: Callable, remote_applicable_builder: RemoteApplicableBuilder
+    ):
+        """Resolves the user specific inputs and creates the applicable.
+        This de-facto allows the user to use a 'remote step()' function.
+        """
+        if not remote_applicable_builder.mutating_fn_kwargs:
+            remote_applicable_builder.mutating_fn_kwargs = self.mutating_function_kwargs
+
+        applicable = step(mutating_fn, **remote_applicable_builder.mutating_fn_kwargs)
+
+        if remote_applicable_builder.when_key_value_pairs is not None:
+            applicable = applicable.when(**remote_applicable_builder.when_key_value_pairs)
+
+        if remote_applicable_builder.when_not_key_value_pairs is not None:
+            applicable = applicable.when_not(**remote_applicable_builder.when_not_key_value_pairs)
+
+        if remote_applicable_builder.when_in_key_value_group_pairs is not None:
+            applicable = applicable.when_in(
+                **remote_applicable_builder.when_in_key_value_group_pairs
+            )
+
+        if remote_applicable_builder.when_not_in_key_value_group_pairs is not None:
+            applicable = applicable.when_not_in(
+                **remote_applicable_builder.when_not_in_key_value_group_pairs
+            )
+
+        if (
+            remote_applicable_builder.name is not None
+            or remote_applicable_builder.namespace is not None
+        ):
+            applicable = applicable.named(
+                name=remote_applicable_builder.name, namespace=remote_applicable_builder.namespace
+            )
+        if remote_applicable_builder.target is not None:
+            applicable = applicable.on_output(target=remote_applicable_builder.target)
+
+        return applicable
+
+    def __call__(self, mutating_fn: Callable):
```

Comment:
We should have some docs as to why this doens't use hte decorator framework -- it's subtle/worth documenting.